utils/helpers/userHelper.ts

```diff
@@ -0,0 +1,53 @@
+import {UserProfile} from "@auth0/nextjs-auth0/src/frontend/use-user";
+import {db} from "../../firebase";
+
+export const getUserFromFirebase = async (user: UserProfile) => {
+    try {
+        //creates user if not created.
+        db.collection("auth0").where('auth0', '==', user.sub).get().then((data) => {
+            if (!data.docs.length) {
+                // to create Auth0 of the new user.
+                db.collection('auth0').doc().set({
+                    auth0: user.sub,
+                }).then(() => {
+                    console.log('auth0 created')
+                })
+                // to create new user.
+                db.collection('users').doc().set({
+                    email: '',
+                    expirationTime: '',
+                    lastSeen: '',
+                    name: user.name,
+                    photoUrl: user.picture,
+                    provider: "Auth0",
+                    resetProfile: true,
+                    uid: user.sub,
+                }).then(() => {
+                    console.log('user created')
+                })
+                // to create profile of the new user.
+                db.collection('profiles').doc().set({
+                    bio: "",
+                    dm: false,
+                    lastName: user.family_name,
+                    location: "",
+                    name: user.name,
+                    profilePic: user.picture,
+                    resetProfile: false,
+                    uid: user.sub,
+                    username: user.nickname
+                }).then(() => {
+                    console.log('profile created')
+                })
+            }
+        })
+
+        //get user
+        const currentUserProfilePromise = db.collection("users").where('uid', '==', user.sub).get();
```

Comment:
should this go inside "then" above? Also with the above changes this is not needed anymore, just query the document with the docRef.id using [v9+ syntax](https://firebase.google.com/docs/firestore/query-data/get-data):

const docRef = doc(db, "users", docRef.id);
const docSnap = await getDoc(docRef);
if (docSnap.exists()) {
  return docSnap.data();
}

Reply:
I believe you should return the profile doc not the user doc.

Reply:
I think we should have functions to retrieve both the user and profile docs. Though, if we're returning only one on log-in, it makes more sense to return the profile.

Reply:
with default data, right?
Now that I think about, the first login ever there is no profile yet. Unless we return the default one.

Reply:
Well it will return whatever the most up-to-date version of the profile is.

First login will be default. But thereafter it will include whatever data users  supplied in the UserProfileForm.